pkg/pdc/client.go

```diff
@@ -35,12 +35,25 @@ type Config struct {
 	HostedGrafanaID string
 	URL             *url.URL
 	RetryMax        int
+
+	// The PDC api endpoint used to sign public keys.
+	// It is not a constant only to make it easier to override the endpoint in local development.
+	SignPublicKeyEndpoint string
+
+	// Used for local development.
+	// Contains headers that are included in each http request send to the pdc api.
+	DevHeaders map[string]string
+
+	// Used for local development.
+	// DevNetwork is the network that the agent will connect to.
+	DevNetwork string
```

Comment:
since `pdc.Client` is an interface, I wonder if it would be cleaner to implement a dev client struct that implements the interface. WDYT?